### DISCUSSION TITLE
Add `tsc` to linting step

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#12787] [Feature Requests]
As @Oreilles correctly pointed out, `vite` doesn't seem to type check the code at the moment. We should update the `lint` command (and therefore the lint CI step) to run `tsc` globally to make sure there's no type regressions introduced

**Comment by oreilles** [comment]
Linting don't check for type error in any of the packages so I don't think adding it to the `app` is the way to go.
Type checking currently only happens during `npm run dev` and `npm run build`, it makes sense to keep it that way.
Suggestion for `app/package.json`:

```diff
- "dev": "vite",
- "build": "vite build",
+ "dev": "run-p dev:*",
+ "dev:tsc": "tsc -w --noEmit",
+ "dev:vite": "vite",
+ "build": "tsc --noEmit && vite build",
```

**Comment by rijkvanzanten** [comment]
> Linting don't check for type error in any of the packages so I don't think adding it to the app is the way to go.

True! I was thinking we'd add it _globally_ on the root of the repo, so we check _every_ bit of TS in the lint step

**Comment by nickrum** [comment]
We'd have to use something like [vue-tsc](https://github.com/johnsoncodehk/vue-tsc) to be able to type-check `*.vue` files.

Doing type-checking as part of the lint step (making it run on each commit) might be a bit heavy.

**Comment by oreilles** [comment]
Agree with @nickrum, linting step should just do linting. Type checking should stay part of the dev/build step imo.

**Comment by rijkvanzanten** [comment]
> Type checking should stay part of the dev/build step imo.

Interesting, I always considered type checking as an additional safety to ensure code quality, not necessarily a build requirement 🤔

**Comment by oreilles** [comment]
If anything type checking feels closer to testing than linting. But it seems that both are accepted. [slate](https://github.com/ianstormtaylor/slate/blob/main/package.json) and [elastic](https://github.com/elastic/eui/blob/master/package.json) does it during linting, [rxjs](https://github.com/ReactiveX/rxjs/blob/master/package.json) and [vue](https://github.com/vuejs/vue-next/blob/master/package.json) during testing.

**Comment by rijkvanzanten** [comment]
@paescuj This something that we could sneak into all the CI updates? 😁

**Comment by paescuj** [comment]
Definitely! I'm going to take a closer look at this 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
